### PR TITLE
[NEW] Rest call to create new channel

### PR DESCRIPTION
**Pull request by aniketsingh03**
Rest call to create new channel has been added. 
Fixes https://github.com/RocketChat/Rocket.Chat.Kotlin.SDK/issues/95

### CODE REVIEW

**Comment by aniketsingh03** [comment]
@luciofm @rafaelks this is ready to be reviewed, please tell me if I might have missed something.

File: core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import chat.rocket.common.model.RoomType
+import chat.rocket.core.RocketChatClient
+import chat.rocket.core.internal.model.CreateNewChannelPayload
+import chat.rocket.core.model.CreateChannelResponse
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.withContext
+import okhttp3.Request
+import okhttp3.RequestBody
+
+suspend fun RocketChatClient.createChannel(roomType: RoomType, name: String, usersList: List<String>?, readOnly: Boolean? = false): CreateChannelResponse = withContext(CommonPool) {
```
Comment (luciofm) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: You should return just the `Room`

You can use `RestResult<Room>` as the `handleRestCall` type, and return `.result()`, take a look at the `signup` method on `Login.kt`

Comment (aniketsingh03) [reply core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: Fine. Can you tell me what should I write in the test for this method to assert that channel was created successfully? Presently I have used the `boolean` variable `status` as an assertion. See https://github.com/aniketsingh03/Rocket.Chat.Kotlin.SDK/blob/fe25994aa4aeaa644ca14a5ec0a93017c9ef0e83/core/src/test/kotlin/chat/rocket/core/internal/rest/ChannelTest.kt#L60

Comment (aniketsingh03) [reply core/src/main/kotlin/chat/rocket/core/internal/rest/Channel.kt]: @luciofm ^

**Comment by aniketsingh03** [comment]
@luciofm can you have a look at this ?

**Comment by aniketsingh03** [comment]
@luciofm @rafaelks can this be merged?

**Comment by rafaelks** [comment]
@luciofm Please, review this again for us.

**Pull request merged**